zoloto/cameras/base.py

```diff
@@ -10,15 +11,17 @@
 from zoloto.marker_dict import MarkerDict
 
 
-class BaseCamera:
-    def __init__(
-        self, *, marker_dict: MarkerDict, calibration_file: Optional[Path] = None
-    ):
-        self.marker_dictionary = cv2.aruco.getPredefinedDictionary(marker_dict)
+class BaseCamera(ABC):
+    def __init__(self, *, calibration_file: Optional[Path] = None):
         self.calibration_file = calibration_file
         self.detector_params = self.get_detector_params(
             cv2.aruco.DetectorParameters_create()
         )
+        self.marker_dictionary = cv2.aruco.getPredefinedDictionary(self.marker_dict)
+
+    @abstractproperty
```

Comment:
This decorator is [deprecated since python 3.3](https://docs.python.org/3/library/abc.html#abc.abstractproperty)
```suggestion
    @property 
    @abstractmethod
```

Reply:
It is, although when I tried this _something_ about it complained, and I couldn't be bothered to fix it right then.

I'll try giving it another go.

Reply:
Changing it doesn't allow the magic abstract property fun this relies on, for reasons I don't entirely understand. Changing to this snippet results in "TypeError: Can't instantiate abstract class TestCamera with abstract methods marker_dict"